query/src/main/java/org/infinispan/query/dsl/embedded/impl/LuceneQueryMaker.java

```diff
@@ -115,6 +116,28 @@ public LuceneQueryMaker(SearchIntegrator searchFactory, FieldBridgeProvider<Type
          isAnalyzerRemote = true;
       }
       Query query = makeQuery(parsingResult.getWhereClause());
+
+      // an all negative top level boolean query is not allowed; needs a bit of rewriting
```

Comment:
This comment was referring to the same case I described earlier. So if all clauses are negative I automatically add #*.* to the query or else it will (wrongly) discard everything.